### ISSUE TITLE
IDE0055 false positive for C#14 extension method

### ISSUE THREAD

**Issue by PhilippNaused** [dotnet/roslyn-analyzers#7654]
### Analyzer

**Diagnostic ID**: [IDE0055](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0055): `Fix formatting`

### Analyzer source

**Version**: 10.0.100-preview.3.25201.16

**NuGet Package**: [Microsoft.CodeAnalysis.CSharp.CodeStyle](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp.CodeStyle)

**Version**: 4.13.0

### Describe the bug

The analyzer raises IDE0055 on the word `extension` and suggests adding a space after it.
That doesn't match the code examples in the docs.
If the extension block has a type parameter, it does the opposite.

### Example code

.editorconfig:
```ini
root = true
[*.cs]
dotnet_diagnostic.IDE0055.severity = warning
```
Test1.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>net8.0</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <LangVersion>preview</LangVersion>
    <NoWarn>CS1591</NoWarn>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeStyle" Version="4.13.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>
</Project>
```
Test.cs
```cs
public static class Extensions
{
    extension(string source) // IDE0055
    {
        public bool IsNullOrEmpty() => string.IsNullOrEmpty(source);
    }

    extension<T>(T? source) // no IDE0055
    {
        public bool IsNull() => source is null;
    }
}
```

**Comment by PhilippNaused** [comment]
Seems to be fixed in `4.14.0`
